gzipi/lib.py

```diff
@@ -146,41 +169,139 @@
 _BYTES_IN_KiB = 1024
 
 
-def _is_valid_gzip_header(gzip_header):
+class Compression(enum.Enum):
+    NONE = None
+    GZIP = 'gzip'
+    ZSTD = 'zstd'
+
+
+SUPPORTED_COMPRESSIONS = {Compression.GZIP.value, Compression.ZSTD.value, }
+
+
+def get_compression_format(path: str) -> Optional[str]:
+    """Get compression format based on the extension."""
+    if not isinstance(path, str):
+        return Compression.NONE.value
+    if path.endswith('.gz'):
+        return Compression.GZIP.value
+    elif path.endswith('.zst'):
+        return Compression.ZSTD.value
+    else:
+        return Compression.NONE.value
+
+
+def _get_compression_type(path: str) -> Optional[str]:
+    with open(path, 'rb') as fin:
+        return _get_compression_type_from_fin(fin)
+
+
+def _get_compression_type_from_fin(fin: IO) -> Optional[str]:
+    header = fin.read(4)
+    fin.seek(0)
+    if header.startswith(_GZIP_HEADER):
+        return Compression.GZIP.value
+    elif header.startswith(_ZSTD_HEADER):
+        return Compression.ZSTD.value
+    return Compression.NONE.value
+
+
+def _sniff_compression_type(path: str, mode: str) -> Optional[str]:
+    compression = get_compression_format(path)
+    if compression:
+        return compression
+    try:
+        compression = _get_compression_type(path) if 'r' in mode else Compression.NONE.value
+        if compression:
+            return compression
+    except Exception as err:
+        _LOGGER.error(err)
+    return compression
+
+
+def _open_file(
+    path: Union[str, IO],
+    mode: str = 'r',
+    compression: Optional[str] = Compression.NONE.value,
```

Comment:
This function will raise ValueError if the default value is passed, right? Perhaps we should make it more obvious that the file must be compressed in one of the supported two formats?